server/internal/server/vector_stores.go

```diff
@@ -0,0 +1,321 @@
+package server
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strconv"
+	"time"
+
+	fv1 "github.com/llm-operator/file-manager/api/v1"
+	"github.com/llm-operator/rbac-manager/pkg/auth"
+	v1 "github.com/llm-operator/vector-store-manager/api/v1"
+	"github.com/llm-operator/vector-store-manager/server/internal/store"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"gorm.io/gorm"
+)
+
+const (
+	vectorStoreObject = "vector_store"
+)
+
+// CreateVectorStore creates a new vector store.
+func (s *S) CreateVectorStore(
+	ctx context.Context,
+	req *v1.CreateVectorStoreRequest,
+) (*v1.VectorStore, error) {
+	userInfo, err := s.extractUserInfoFromContext(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	if req.Name == "" {
+		return nil, status.Errorf(codes.InvalidArgument, "name is required")
+	}
+
+	// Pass the Authorization to the context for downstream gRPC calls.
+	ctx = auth.CarryMetadata(ctx)
+
+	for _, fid := range req.FileIds {
+		if err := s.validateFile(ctx, fid); err != nil {
+			return nil, err
+		}
+	}
+
+	c, err := s.store.GetCollectionByName(userInfo.ProjectID, req.Name)
+	if err == nil {
+		return nil, status.Errorf(codes.AlreadyExists, "vector store %q already exists", req.Name)
+	}
+	if !errors.Is(err, gorm.ErrRecordNotFound) {
+		return nil, status.Errorf(codes.Internal, "get collection: %s", err)
+	}
+
+	cid, err := s.vstoreClient.CreateVectorStore(ctx, req.Name)
+	if err != nil {
+		return nil, err
+	}
+
+	c = &store.Collection{
+		CollectionID:   cid,
+		Name:           req.Name,
+		Status:         store.CollectionStatusInProgress,
+		OrganizationID: userInfo.OrganizationID,
+		ProjectID:      userInfo.ProjectID,
+		TenantID:       userInfo.TenantID,
+		LastActiveAt:   time.Now().Unix(),
+	}
+	if req.ExpiresAfter != nil {
+		c.Anchor = store.ExpiresAfterAnchor(req.ExpiresAfter.Anchor)
+		c.ExpiresAfterDays = req.ExpiresAfter.Days
+	}
+	if err := s.store.CreateCollection(c); err != nil {
```

Comment:
Could you add a TODO comment on creating a collection and metadata in a single DB transaction?